extern/sector-storage/manager.go

```diff
@@ -273,16 +273,24 @@ func (m *Manager) SectorsUnsealPiece(ctx context.Context, sector storage.SectorR
 	defer cancel()
 
 	log.Debugf("acquire unseal sector lock for sector %d", sector.ID)
-	if err := m.index.StorageLock(ctx, sector.ID, storiface.FTSealed|storiface.FTCache, storiface.FTUnsealed); err != nil {
+	if err := m.index.StorageLock(ctx, sector.ID, storiface.FTSealed|storiface.FTCache|storiface.FTUpdate|storiface.FTUpdateCache, storiface.FTUnsealed); err != nil {
 		return xerrors.Errorf("acquiring unseal sector lock: %w", err)
 	}
 
 	// if the selected worker does NOT have the sealed files for the sector, instruct it to fetch it from a worker that has them and
 	// put it in the sealing scratch space.
 	sealFetch := func(ctx context.Context, worker Worker) error {
 		log.Debugf("copy sealed/cache sector data for sector %d", sector.ID)
-		if _, err := m.waitSimpleCall(ctx)(worker.Fetch(ctx, sector, storiface.FTSealed|storiface.FTCache, storiface.PathSealing, storiface.AcquireCopy)); err != nil {
-			return xerrors.Errorf("copy sealed/cache sector data: %w", err)
+		_, err := m.waitSimpleCall(ctx)(worker.Fetch(ctx, sector, storiface.FTSealed|storiface.FTCache, storiface.PathSealing, storiface.AcquireCopy))
+		if err != nil && !xerrors.Is(err, storiface.ErrSectorNotFound) {
```

Comment:
Those errors won't get propagated, would need https://github.com/filecoin-project/go-jsonrpc/pull/36 (Maybe worth finally landing for this?)

Reply:
@magik6k I found the issue that was causing these errors to be returned as strings and it is fixed in the latest commit. Is there still a codepath where this might hit RPC? If so, could you show me where it is?

Reply:
worker.Fetch is an rpc call.

This will only work with the miner built-in worker, as that one doesn't go through rpc.

Reply:
@magik6k Talked this over with Zen and we decided that not checking the error type will still give us the desired result in all cases. If you agree, I think we can land this PR